projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts

```diff
@@ -194,6 +195,18 @@ export class IgxHierarchicalGridNavigationService extends IgxGridNavigationServi
         }
     }
 
+    public onKeydownArrowRight(element, rowIndex, visibleColumnIndex, isSummary = false) {
```

Comment:
I don’t see a need to add additional parameters and overwrite the onKeydownArrowRight, onKeydownArrowLeft, onKeydownHome methods.
This will also not cover all possible scenarios where the grid may flicker on IE as if you make a global search and check where this.grid.nativeElement.focus({ preventScroll: true }) is used you’ll see that it is called in other functions as well (in the base grid navigation and also in the hierarchical grid navigation).

The way it was implemented in the first commit to this PR was better. Just apply the changes suggested there so that each grid implements its own focusGridWithoutScrolling method in their related navigation service, move the hierarchical grid specific code in the method of the hierarchical grid service and make sure to replace all instances where this.grid.nativeElement.focus({ preventScroll: true }) is used with the this.focusGridWithoutScrolling() method call.

For example:

In grid-navigation.service.ts

protected focusGridWithoutScrolling() {
	this.grid.nativeElement.focus({ preventScroll: true });
}

In hierarchical-grid-navigation.service.ts

protected focusGridWithoutScrolling() {
	if(isIE()) {
		this.grid.rootGrid.nativeElement.focus();
} else {
	super.focusGridWithoutScrolling();
}
}